cfn-resources/stream-connection/cmd/resource/mappings_test.go

```diff
@@ -263,6 +263,38 @@ func TestGetStreamConnectionClusterTypeModel(t *testing.T) {
 	})
 }
 
+func TestGetStreamConnectionSampleTypeModel(t *testing.T) {
+	streamsConnSample := &admin.StreamsConnection{
+		Name: ptr.String("sample_stream_solar"),
+		Type: ptr.String("Sample"),
+	}
+	t.Run("With Nil Current Model", func(t *testing.T) {
+		result := resource.GetStreamConnectionModel(streamsConnSample, nil)
+
+		assert.NotNil(t, result)
+		assert.Equal(t, *streamsConnSample.Name, *result.ConnectionName)
+		assert.Equal(t, *streamsConnSample.Type, *result.Type)
+		assert.Nil(t, result.DbRoleToExecute)
+	})
+
+	t.Run("Sample Stream Solar dataset", func(t *testing.T) {
+		currentModel := &resource.Model{
+			Profile:        ptr.String("default"),
+			ProjectId:      ptr.String("testProjectID"),
+			InstanceName:   ptr.String("TestInstance"),
+			ConnectionName: ptr.String("sample_stream_solar"),
+			Type:           ptr.String("Sample"),
+		}
+		result := resource.GetStreamConnectionModel(streamsConnSample, currentModel)
+
+		assert.Equal(t, *currentModel.InstanceName, *result.InstanceName)
```

Comment:
knit: i think this is ok.  when you use many assert in a test you may consider:
  assert := assert.New(t)
and then
assert.Equal(*currentModel.Profile, *result.Profile)
etc. ( you don't need to pass t every time )

Reply:
Thanks for the tip. I was following the earlier test cases. But refactored now 👍